libs/cache/single_pointer_cache.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SINGLE_POINTER_CACHE_HPP
+#define IROHA_SINGLE_POINTER_CACHE_HPP
+
+#include <memory>
+#include <mutex>
+
+namespace iroha {
+  namespace cache {
+
+    /**
+     * Thread-safely stores and returns shared pointer to an element of template
+     * type
+     */
+    template <typename DataType>
+    class SinglePointerCache {
+      using DataPointer = std::shared_ptr<DataType>;
```

Comment:
please add private specifier explicitly.

Also why using is private? Users might need it.